src/gt4py/stencil_builder.py

```diff
@@ -125,6 +137,29 @@ def with_backend(self: "StencilBuilder", backend_name: str) -> "StencilBuilder":
         self.backend = gt4py.backend.from_name(backend_name)(self)
         return self
 
+    @classmethod
+    def default_options_dict(
+        cls, definition_func: Union[StencilFunc, AnnotatedStencilFunc]
+    ) -> Dict[str, Any]:
+        return {"name": definition_func.__name__, "module": definition_func.__module__}
+
+    @classmethod
+    def name_to_options_args(cls, name: Optional[str]) -> Dict[str, str]:
+        if not name:
+            return {}
+        components = name.rsplit(".")
+        data = {"name": name}
+        if len(components) > 1:
+            data = {"module": components[0], "name": components[1]}
+        return data
+
+    @classmethod
+    def nest_impl_options(cls, options_dict: Dict[str, Any]) -> Dict[str, Any]:
+        impl_opts = options_dict.setdefault("impl_opts", {})
+        for impl_key in (k for k in options_dict.keys() if k.startswith("_")):
+            impl_opts[impl_key] = options_dict.pop(impl_key)
```

Comment:
```suggestion
        impl_opts = {k: v for k, v in options_dict.items() if k.startswith("_")}
```

Reply:
Possibly this requires a comment. Note that your suggestion will leave the `_something` keys in `options_dict`. The BuildOption constructor will fail on these.